code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -974,10 +974,53 @@
 			else
 				message = "makes a light spitting noise, a poor attempt at a whistle."
 
+		if ("roll")
+			if(!param)
+				to_chat(src, "To use this emote specify what die you want to roll in the first line. Prefix it with a \"D\". You may add a bonus to your roll, simply add a \"+\" and your number after your chosen die. If you want to incorporate a difficulty check, add a \"-DC\". You may use a bonus, a difficulty check or both.<br>Here is an example: roll-D20+5-DC15. This will roll a twenty-sided die, add five and compare it against a difficulty check of 15.")
+				return
+			else
+				var/t1 = findtext(param, "+", 1, null)
+				var/t2 = findtext(param, "-", 1, null)
+				var/die = copytext(param, 1, t1)
+				var/bonus = copytext(param, t1 + 1, t2)
+				var/dc = copytext(param, t2 + 1, length(param) + 1 )
+				if (!t1)
+					bonus = null
+				if (!t2)
+					dc = null
+				var/die_number = text2num(copytext(die, 2, length(die) + 1))
+				var/bonus_number = text2num(bonus)
+				var/dc_number = text2num(copytext(dc, 3, length(dc) + 1))
+
+				var/die_result = rand(die_number)
+				var/die_total = die_result + bonus_number
+
+				if (die_number)
```

Comment:
this should be structured in an else if chain, starting with the "die_number && bonus_number && dc_number" check etc.